python-package/lightgbm/dask.py

```diff
@@ -110,12 +110,130 @@ def _train_part(
+                    continue
+
+                eval_set = part['eval_set'][i]
+                if eval_set == '__train__':
+                    x_e.append(part['data'])
+                    y_e.append(part['label'])
+                else:
+                    x_e.extend(eval_set[0])
+                    y_e.extend(eval_set[1])
+
+                eval_weight = part.get('eval_sample_weight')
+                if eval_weight:
+                    if eval_weight[i] == '__sample_weight__':
+                        w_e.append(part['weight'])
+                    else:
+                        w_e.extend(eval_weight[i])
+
+                eval_group = part.get('eval_group')
+                if eval_group:
+                    if eval_group[i] == '__group__':
+                        g_e.append(part['group'])
+                    else:
+                        g_e.extend(eval_group[i])
+
+                eval_class_weight = part.get('eval_class_weight')
+                if eval_class_weight:
+                    if len(eval_class_weight) > len(local_eval_class_weight):
+                        local_eval_class_weight = eval_class_weight
+
+                eval_init_score = part.get('eval_init_score')
+                if eval_init_score:
+                    if eval_init_score[i] == '__init_score__':
+                        init_score_e.append(part['init_score'])
+                    else:
+                        init_score_e.extend(eval_init_score[i])
+
+                eval_names = part.get('eval_names')
+                if eval_names:
+                    if len(eval_names) > len(local_eval_names):
+                        local_eval_names = eval_names
+
+            # _concat each eval component.
+            local_eval_set.append((_concat(x_e), _concat(y_e)))
+            if w_e:
+                local_eval_sample_weight.append(_concat(w_e))
+            if init_score_e:
+                local_eval_init_score.append(_concat(init_score_e))
+            if g_e:
+                local_eval_group.append(_concat(g_e))
+
     try:
         model = model_factory(**params)
         if is_ranker:
-            model.fit(data, label, sample_weight=weight, init_score=init_score, group=group, **kwargs)
+            model.fit(
+                data,
+                label,
+                sample_weight=weight,
+                init_score=init_score,
+                group=group,
+                eval_set=local_eval_set,
+                eval_sample_weight=local_eval_sample_weight,
+                eval_init_score=local_eval_init_score,
+                eval_group=local_eval_group,
+                eval_names=local_eval_names,
+                **kwargs
+            )
         else:
-            model.fit(data, label, sample_weight=weight, init_score=init_score, **kwargs)
+            if 'eval_at' in kwargs:
+                kwargs.pop('eval_at')
+            if local_eval_class_weight:
+                kwargs['eval_class_weight'] = local_eval_class_weight
```

Comment:
I'm confused about this one, since in the `.fit()` call below you already pass `eval_sample_weight=local_eval_sample_weight`. If think if this condition was true, you'd get an error like "SyntaxError: keyword argument repeated" or something.

Does anything bad happen if you pass `eval_sample_weight=None` into `.fit()`? I think it should be fine, and if so you can remove this condition.

Reply:
I think my concern here was that `model` could be referring to a regressor or a classifier. `eval_class_weight` when provided to `LGBMRegressor.fit` throws the unexpected keyword argument error.

Reply:
ah got it, ok makes sense!